packages/@react-aria/collections/src/CollectionBuilder.tsx

```diff
@@ -127,22 +127,40 @@ function useCollectionDocument<T extends object, C extends BaseCollection<T>>(cr
 
 const SSRContext = createContext<BaseNode<any> | null>(null);
 
-function useSSRCollectionNode<T extends Element>(Type: string, props: object, ref: ForwardedRef<T>, rendered?: any, children?: ReactNode, render?: (node: Node<T>) => ReactElement) {
+export type CollectionNodeClass<T> = {
+  new (key: Key): CollectionNode<T>,
+  readonly type: string
```

Comment:
this is readonly? but it's not readonly because you assign to it just below?

Reply:
it is readonly, the `createCollectionNodeClass` below basically creates a CollectionNodeClass when called since you can't set the type after that call

Reply:
ah, confusing...
Something is still tripping me up here, like inheritance isn't quite how I would expect or something